### PR TITLE
Deprecate Homebrew/homebrew-command-not-found

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
This doesn't seem to be used at all nowadays:
https://formulae.brew.sh/analytics/brew-command-run/365d/

As an official tap, it has overhead on e.g. RuboCop upgrades and a presence in Homebrew/brew CI, manpage, etc. that no longer seems to be warranted.

If there's enough 👍🏻 to do this: I'll add various `odeprecated` or `odisabled` calls to Homebrew/homebrew-command-not-found and archive the repository.

### CODE REVIEW

**Comment by bevanjkay** [comment]
I don't think this would show on the command-run analytics, because it's not invoked directly.
Unless we're looking for the `command-not-found-init` command, which may not be used as opposed to reading the docs.

In saying that, I've personally never used it.

**Comment by MikeMcQuaid** [comment]
> I don't think this would show on the command-run analytics, because it's not invoked directly.

I guess my thinking is: none of the relevant commands show up in the analytics.

> In saying that, I've personally never used it.

This is an added datapoint: from a show of hands at the last Homebrew AGM iirc no maintainers used this.

**Comment by carlocab** [comment]
> This is an added datapoint: from a show of hands at the last Homebrew AGM iirc no maintainers used this.

I remember a few people raising their hands (including me...)

**Comment by MikeMcQuaid** [comment]
> I remember a few people raising their hands (including me...)

@carlocab My bad, maybe it was homebrew-autoupdate people didn't use?

Any idea why there's no results showing up in analytics here? Would you be happy to take on any improvements to e.g. not committing data to repo without PRs?

**Comment by carlocab** [comment]
> @carlocab My bad, maybe it was homebrew-autoupdate people didn't use?

Yes, that one I recall no one raising their hands for.

> Any idea why there's no results showing up in analytics here? 

No 🥲 

> Would you be happy to take on any improvements to e.g. not committing data to repo without PRs?

Yep, happy to take on improvements, but I'd like some assistance with continuing maintenance (even if it's just poking me on Slack to fix things).

**Comment by MikeMcQuaid** [comment]
> > Any idea why there's no results showing up in analytics here?
> 
> No 🥲

@carlocab Which commands do you manually run and when, just so I understand?



> Yep, happy to take on improvements, but I'd like some assistance with continuing maintenance (even if it's just poking me on Slack to fix things).

Yup, fair.

Will leave this open for now in case anyone else feels strongly it shouldn't be part of Homebrew.

**Comment by cho-m** [comment]
> Any idea why there's no results showing up in analytics here?

I think we only collect analytics for the new `AbstractCommand` subclasses?
https://github.com/Homebrew/brew/blob/92adf6a0b71e59d3348c982d11e2560dc9317687/Library/Homebrew/brew.rb#L88-L91

`Homebrew/homebrew-command-not-found` (and probably other official tap commands) have not been refactored.

**Comment by MikeMcQuaid** [comment]
Good catch, thanks @cho-m!

**Comment by carlocab** [comment]
> @carlocab Which commands do you manually run and when, just so I understand?

I rarely run it manually these days. When I do, it's usually to run `brew which-formula` because I want to know which formula provides a command.

However, I do have it set up as a command-not-found handler, via https://github.com/ohmyzsh/ohmyzsh/tree/5e59d21e5465ec0694441f41ff0acf1ddf4de187/plugins/command-not-found.

**Review by Rylan12 — APPROVED** [review]
I use this too, but only the `which-formula` command (which I've aliased to `brew which`) and not the shell handler part.

However, I don't use it frequently enough that I think it needs to be official. If we deprecate it here, I'll likely keep a personal fork of it around for my personal use

**Comment by MikeMcQuaid** [comment]
> Would it be worth updating the command to the use the new DSL so we can collect stats before deprecating it?

Yes, I think this is the best thing for now.

Going to mark this as draft for now until we've done that.

**Comment by reitermarkus** [comment]
I'm also using this. And of course, since most of the software I use is already installed by my dotfiles, `command-not-found` is very rarely actually called. `command-not-found-init` shouldn't do any network request given it's called when initializing a new terminal like `shellenv`.

**Comment by MikeMcQuaid** [comment]
Have opened PRs to gather analytics: https://github.com/Homebrew/homebrew-command-not-found/pull/156 (and others). Closing this until at least 30 days after merge so we can see what usage levels are like.

**Pull request closed without merging**
